support/k8s/k8s-workload-registrar/controllers/spiffeid_controller.go

```diff
@@ -0,0 +1,295 @@
+		Selectors: []*common.Selector{
+			{Type: "k8s_psat", Value: fmt.Sprintf("cluster:%s", r.c.Cluster)},
+		},
+	})
+}
+
+// Reconcile ensures the SPIRE Server entry matches the corresponding CRD
+func (r *SpiffeIDReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	spiffeID := spiffeidv1beta1.SpiffeID{}
+
+	if err := r.Get(ctx, req.NamespacedName, &spiffeID); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			r.c.Log.WithError(err).Error("unable to fetch SpiffeID")
+			return ctrl.Result{}, err
+		}
+
+		// Delete event
+		if err := r.ensureDeleted(ctx, r.spiffeIDCollection[req.NamespacedName.String()]); err != nil {
+			r.c.Log.WithFields(logrus.Fields{
+				"entryid": r.spiffeIDCollection[req.NamespacedName.String()],
+			}).WithError(err).Error("unable to delete spire entry", )
+			return ctrl.Result{}, err
+		}
+		delete(r.spiffeIDCollection, req.NamespacedName.String())
+		r.c.Log.WithFields(logrus.Fields{
+			"entry": req.NamespacedName.String(),
+		}).Info("Finalized entry")
+
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	entryId, err := r.updateOrCreateSpiffeID(ctx, &spiffeID)
+	if err != nil {
+		r.c.Log.WithFields(logrus.Fields{
+			"request": req,
+		}).WithError(err).Error(err, "unable to get or create spire entry")
+		return ctrl.Result{}, err
+	}
+
+	oldEntryId := spiffeID.Status.EntryId
+	if oldEntryId == nil || *oldEntryId != entryId {
+		// We need to update the Status field
+		if oldEntryId != nil {
+			// entry resource must have been modified, delete the hanging one
+			if err := r.ensureDeleted(ctx, *spiffeID.Status.EntryId); err != nil {
+				r.c.Log.WithFields(logrus.Fields{
+					"entryid": *spiffeID.Status.EntryId,
+				}).WithError(err).Error("unable to delete old spire entry")
+				return ctrl.Result{}, err
+			}
+		}
+		spiffeID.Status.EntryId = &entryId
+		if err := r.Status().Update(ctx, &spiffeID); err != nil {
+			r.c.Log.WithError(err).Error("unable to update SpiffeID status")
+			return ctrl.Result{}, err
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// ensureDeleted deletes the specified entry on the SPIRE Server
+func (r *SpiffeIDReconciler) ensureDeleted(ctx context.Context, entryId string) error {
+	if _, err := r.c.R.DeleteEntry(ctx, &registration.RegistrationEntryID{Id: entryId}); err != nil {
+		if status.Code(err) != codes.NotFound {
+			if status.Code(err) == codes.Internal {
+				// Spire server currently returns a 500 if delete fails due to the entry not existing.
+				// We work around it by attempting to fetch the entry, and if it's not found then all is good.
+				if _, err := r.c.R.FetchEntry(ctx, &registration.RegistrationEntryID{Id: entryId}); err != nil {
+					if status.Code(err) == codes.NotFound {
+						r.c.Log.WithFields(logrus.Fields{
+							"entry": entryId,
+						}).Info("Entry already deleted")
+						return nil
+					}
+				}
+			}
+			return err
+		}
+	}
+	r.c.Log.WithFields(logrus.Fields{
+		"entry": entryId,
```

Comment:
We should be consistent on this field name "entryid" v.s. "entry" v.s. "entry-id"

Reply:
Changed to use `EntryID`.